RELEASE.md

```diff
@@ -44,9 +44,9 @@ The operator should be released within a week after the [OpenTelemetry collector
 
 | Version  | Release manager |
 |----------|-----------------|
-| v0.102.0 | @swiatekm-sumo  |
 | v0.103.0 | @frzifus        |
 | v0.104.0 | @jaronoff97     |
 | v0.105.0 | @pavolloffay    |
 | v0.106.0 | @yuriolisa      |
-| v0.107.0 | @TylerHelmuth   |
\ No newline at end of file
+| v0.107.0 | @TylerHelmuth   |
+| v0.102.0 | @swiatekm-sumo  |
```

Comment:
If you're going to steal this release from me, you should keep me at the top of the list and move yourself to the bottom imo.